packages/server/graphql/mutations/helpers/notifications/MattermostNotifier.ts

```diff
@@ -0,0 +1,339 @@
+import formatTime from 'parabol-client/utils/date/formatTime'
+import formatWeekday from 'parabol-client/utils/date/formatWeekday'
+import makeAppURL from 'parabol-client/utils/makeAppURL'
+import findStageById from 'parabol-client/utils/meetings/findStageById'
+import {phaseLabelLookup} from 'parabol-client/utils/meetings/lookups'
+import appOrigin from '../../../../appOrigin'
+import Meeting from '../../../../database/types/Meeting'
+import {SlackNotificationEventEnum as EventEnum} from '../../../../database/types/SlackNotification'
+import {IntegrationProviderMattermost} from '../../../../postgres/queries/getIntegrationProvidersByIds'
+import {toEpochSeconds} from '../../../../utils/epochTime'
+import MattermostServerManager from '../../../../utils/MattermostServerManager'
+import segmentIo from '../../../../utils/segmentIo'
+import sendToSentry from '../../../../utils/sendToSentry'
+import {DataLoaderWorker} from '../../../graphql'
+import getSummaryText from './getSummaryText'
+import {
+  makeFieldsAttachment,
+  makeHackedButtonPairFields,
+  makeHackedFieldButtonValue
+} from './makeMattermostAttachments'
+import {NotificationIntegrationHelper} from './NotificationIntegrationHelper'
+import {Notifier} from './Notifier'
+
+const notifyMattermost = async (
+  event: EventEnum,
+  webhookUrl: string,
+  userId: string,
+  teamId: string,
+  textOrAttachmentsArray: string | unknown[],
+  notificationText?: string
+) => {
+  const manager = new MattermostServerManager(webhookUrl)
+  const result = await manager.postMessage(textOrAttachmentsArray, notificationText)
+  if (result instanceof Error) {
+    sendToSentry(result, {userId, tags: {teamId, event, webhookUrl}})
+    return {
+      error: result
+    }
+  }
+  segmentIo.track({
+    userId,
+    event: 'Mattermost notification sent',
+    properties: {
+      teamId,
+      notificationEvent: event
+    }
+  })
+
+  return 'success'
+}
+
+const makeEndMeetingButtons = (meeting: Meeting) => {
```

Comment:
This will work locally, but if we only pass `meetingId` to the notifier, it needs to load the meeting and will only get `Meeting` as type. At some point we would need to make sure it's of type `AnyMeeting` which would require basically the same check.